src/Model.php

```diff
@@ -456,7 +470,18 @@ public function set($field, $value = null)
 
         $f = $this->hasElement($field);
 
-        // set hook here
+        try {
+            if ($this->hook('normalize') !== false && $f) {
+                $value = $f->normalize($value);
+            }
+        } catch (Exception $e) {
+            if (method_exists($e, 'addMoreInfo')) {
```

Comment:
Let's add correct atk4/core version dependency in composer.json and then we can get rid of this condition.